### PR TITLE
New image witout pyenv and upgraded ubuntu to 16.04 + docker client

### PR DESCRIPTION
**Pull request by jeduden**

### CODE REVIEW

File: py27_py35/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    && rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update \
+    && apt-get install time netcat gettext libjpeg-dev curl git-core build-essential \
+         make build-essential libssl-dev zlib1g-dev libbz2-dev \
+         libreadline-dev libsqlite3-dev -y \
+    && rm -rf /var/lib/apt/lists/*
+
+
+RUN curl https://bootstrap.pypa.io/get-pip.py > /tmp/get-pip.py && for pybin in $pythonVersions; do $pybin /tmp/get-pip.py; done && rm -rf /tmp/get-pip.py
```
Comment (sergray) [inline py27_py35/Dockerfile]: It is more secure to COPY local get-pip.py

File: py27_py35/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+RUN apt-get update \
```
Comment (sergray) [inline py27_py35/Dockerfile]: May it be better to reuse updated repositories and clean up them later?

File: py27_py35/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    && apt-get install -y \
+         apt-transport-https \
+         ca-certificates \
+         software-properties-common \
+    && curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add - \
+    && add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" \
+    && apt-get update \
+    && apt-get install -y docker-ce
+
+RUN curl -L https://github.com/docker/compose/releases/download/1.12.0/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose && chmod +x /usr/local/bin/docker-compose
```
Comment (sergray) [inline py27_py35/Dockerfile]: What about having it locally?

**Review by sergray — CHANGES_REQUESTED** [review]

**Comment by jeduden** [comment]
@sergray comments addressed.

**Pull request merged**